src/config.rs

```diff
@@ -155,9 +155,19 @@ pub fn get_log_setting(config_contents: &str) -> bool {
 pub fn get_hook_script(config_contents: &str, hook_name: &str) -> Result<String, String> {
     match get_table_key_value_from_config(config_contents, "hooks", hook_name) {
         Err(err) => Err(err),
-        Ok(value) => match value.as_str() {
-            Some(script) => Ok(String::from(script)),
-            None => Err(String::from("Invalid hook config")),
+        Ok(value) => match value {
+            Value::String(script) => Ok(script),
+            Value::Array(script) => {
+                let script = script.iter().map(|f| f.as_str());
+                match script.clone().any(|f| f.is_none()) {
```

Comment:
This `.clone()` has to be made because the `.any()` consumes the script iterator. I suppose there is a better way to do this. I'm gonna research some more and see if I can make it more ergonomic, or is this fine?

Reply:
A clone here wouldn't typically be very expensive, but I think we can avoid it by leveraging some additional iterator power

Could you try something like this and see if it works (also includes a suggested error message tweak, and rustfmt will definitely need to adjust):

```rust
Value::Array(val) => Ok(val
    .iter()
    .map(|v| v.as_str())
    .collect::<Option<Vec<_>>>()
    .ok_or(format!("Invalid hook config for {}. An element in the array is not a string", hook_name))?
    .join(" && ")),
```

Since `as_str()` returns an `Option`, we can take advantage of the fact that `Option` implements `FromIterator` by adjusting the type in `collect`. That way the iterator will break early on the first non-string item found and spare us a few iterations. We can then use `ok_or` to map the Option to a `Result` with an early exit (`?`) should it be an `Error` (a non-string was found)

Reply:
I always forget how powerful `.collect()` is. This suggestion worked without changing anything :D Thank you for the thorough explanation and for teaching me "the Rust way"!